sympy/solvers/bivariate.py

```diff
@@ -176,22 +185,95 @@ def _lambert(eq, x):
     p, den = den.as_coeff_Mul()
     e = exp(num/den)
     t = Dummy('t')
-    args = [d/(a*b)*t for t in roots(t**p - e, t).keys()]
-
     # calculating solutions from args
-    for arg in args:
+    if p == 1:
+        t = e
+        args = [d/(a*b)*t]
+    elif domain is S.Reals:
+        ind_ls = [d/(a*b)*t for t in roots(t**p - e, t).keys()]
+        args = []
+        j = -1
+        for i in ind_ls:
+            j += 1
+            if not isinstance(i,int):
+                if not i.has(I):
+                    args.append(ind_ls[j])
+            elif isinstance(i,int):
+                args.append(ind_ls[j])
+    else:
+        args = [d/(a*b)*t for t in roots(t**p - e, t).keys()]
+    if len(args) == 0:
+        return S.EmptySet
+    # for simplicity defined two functions
+    # Lambert_Real_Braches , lambert_helper
+    def lambert_braches():
         for k in lambert_real_branches:
-            w = LambertW(arg, k)
-            if k and not w.is_real:
+            rhs2 = -c/b + (a/d)*LambertW(arg, k)
+            if rhs2.has(I):
                 continue
-            rhs = -c/b + (a/d)*w
+            sol1 = lambert_helper(rhs2)
+        return sol1, rhs2
+
+    def lambert_helper(rhs):
+    # to avoid repetitive code
+        for xu in xusolns:
+            sol.append(xu.subs(u, rhs))
+        return sol
 
-            for xu in xusolns:
-                sol.append(xu.subs(u, rhs))
-    return sol
+    n = Dummy('n',integer=True)
+    for arg in args:
+        rhs1 = -c/b + (a/d)*LambertW(arg)
+        lambert_real = LambertW(arg).is_real
+
+        if domain_real or domain is None:
+            if domain_real:
+                if not len(list(eq.atoms(Symbol,Dummy))) >1:
+                    if lambert_real:
+                        if -1/E <= re(arg) < 0:
+                            sol,rhs = lambert_braches()
+                        elif re(arg) >= 0:
+                            rhs = rhs1
+                    elif re(arg) < -1/E:
+                        return S.EmptySet
+
+                if (arg.has(Symbol)) and not arg.has(Dummy):
+                    SymbolStatus = list(arg.atoms(Symbol))[0]
```

Comment:
SymbolStatus is not a good variable name. Capitalised names are used for class names. A variable should have a lower-case name unless it is a mathematical symbol like `M`.